### ISSUE TITLE
[Demand] Android native QR code scan feature

### ISSUE THREAD

**Issue by jk234ert** [DimensionDev/Maskbook#2039]
## Catalogue

- [x] UI/UX
- [ ] Maskbook Core Feature
- [ ] Maskbook Plugin Feature
- [ ] Security/Cryptography Feature
- [ ] Other
  - /_List it here_/

## Part 1: Introduction and Background
For a better user experience in scanning QR code, we need to implement the Android native QR code scan feature and integrate it into the new GeckoView messaging channel to make it work with JS side

## Part 2: Problem to Solve
The bad user experience of current WebRTC based QR code scan

## Notes
Tow sub-tasks involved:
1. Implement Android native QR scan
2. Integrate it with the new GeckoView messaging channel

**Comment by jk234ert** [comment]
Draft design:
1. JS side:
Add a new `WebAPI`: `scanQRCode(): Promise<String>`, implement as following:
    a. When JS side need to trigger a new `scanQRCode` session, JS invokes `scanQRCode()`, then waits for the response from native.
    b. When JS receives a String value, it tries to decode it and performs "importing personas", if successfully, navigates to the proper location, otherwise, displays a popup error with error message.

2. Android native side:
When receives the `scanQRCode` message from JS side, it shows the QR code scan view, then send back the result by sending a new message to GeckoView's port

**Comment by Jack-Works** [comment]
Does this function matches iOS version `scanQRCode`?

**Comment by jk234ert** [comment]
> Does this function matches iOS version `scanQRCode`?

Yes, `iOSNativeAPIs ` already has a `scanQRCode(): Promise<string>`

**Comment by maxyou** [comment]
The QR scan code of Android native side is ready.
